client/src/views/ResultsView.vue

```diff
@@ -79,22 +79,25 @@ const categories = [
 ];
 const activeCategory = categories.find((e) => e.name === props.category);
 
+// Name the window
+watchEffect(() => {
+  setWindowName(activeCategory.title);
+});
+
 const {
   fetchData,
   data: results,
-  // errorMessage,
+  changeApiExtension,
 } = useData(ApiService.getResults, activeCategory.apiExtensionString);
 
-// Name the window
 watchEffect(() => {
-  setWindowName(activeCategory.title);
+  changeApiExtension(activeCategory.apiExtensionString, router.params);
```

Comment:
Wir können auch jeder Wertung ein eigenes view spendieren. Das composable wird immer komplizierter und ich habe mich in letzter Zeit oft dabei ertappt dass ich nicht wisse, welche Wertung jetzt eigentlich in welchem view ist. Generische Komponenten sind ja cool aber ich habe das Gefühl, dass wird so langsam etwas unübersichtlich. Es wird nicht viel mehr Code, aber einfacher zu verstehen.

Reply:
Auch das würden wir uns damit ersparen:
<img width="826" alt="Screenshot 2022-01-06 at 05 47 02" src="https://user-images.githubusercontent.com/35194489/148378528-1aa4d432-db58-444c-a075-e6bea5dbd2d8.png">
Nach dem reload gehts dann. Nicht aber wenn man nur dahin navigiert. Kann man natürlich fixen aber bestärkt mich nur in der Meinung, das wir einfach jedem eine eigene view geben sollten. Das scheint auch best practice zu sein.

Reply:
Hatte Gestern Abend eigentlich auch den gleichen Gedanken.

Wenn man nur um ein paar Zeilen Code sparen kann, im Gegensatz die Komplexität drastisch erhöht, ist es das definitiv nicht wert.

Baue ich gleich um.

Reply:
Aber wundert mich, dass es beim navigieren nicht funktioniert. Genau deshalb hatte ich das ja geändert. Darüber ist es mir ja aufgefallen und so hatte ich es auch getestet.
Das es bei einem Reload funktioniert ist logisch, da da die Ganze View neugeladen wird und das Composable mit dem richtigen Endpoint geladen wird.